### ISSUE TITLE
PHP 7.4 / YAF 3.0.9 DLL / Building error

### ISSUE THREAD

**Issue by MrJackDuaner** [laruence/yaf#464]
Good afternoon,

I am currently using php-7.3.6-nts-Win32-vc15-x64 with version 3.0.8 of php_yaf.dll.
I would like to use php-7.4.3-nts-Win32-vc15-x64 and I was looking for YAF DLL compatible with this version.
I found that version 3.0.9 of YAF was OK for running with PHP 7.4, but I didn't find DLL file on page https://pecl.php.net/package/yaf/3.0.9/windows , only versions for PHP 7.3 and PHP 7.2.
So I decided to build YAF DLL from sources.

As I have never built PHP / DLL extension, I have followed procedure from https://wiki.php.net/internals/windows/stepbystepbuild_sdk_2 .
For building php-7.4.3-nts-Win32-vc15-x64, I have downloaded / installed :
- Visual Studio Community 2017 (version 15.0) : https://my.visualstudio.com/Downloads?q=Visual%20Studio%20Community%202017%20(version%2015.0)&pgroup=
- PHP SDK 2.2 : https://github.com/Microsoft/php-sdk-binary-tools
- PHP 7.4.3 sources : https://windows.php.net/downloads/releases/php-7.4.3-src.zip
- YAF master sources : https://pecl.php.net/get/yaf-3.0.9.tgz or https://github.com/laruence/yaf

Building PHP 7.4.3 is working fine.
But when building YAF from 3.0.9 sources (pecl.php.net) or master sources (github.com), I got the following output (just the last logs containing the error) :

`yaf_session.obj : error LNK2019: unresolved external symbol php_session_start referenced in function zim_yaf_session___destruct@@16
C:\php-sdk-binary-tools-master\phpdev\vc15\x64\php-7.4.3-src\x64\Release_TS\php7ts.dll : fatal error LNK1120: 1 unresolved externals
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX64\x64\link.exe"' : return code '0x460'
Stop.`

I have searched for a reason / an answer of this error on the Web but didn't find anything that could solve the problem.

Do you have an idea about this issue ?

Or may be will you publish DLLs for PHP 7.4 on YAF 3.0.9 page ?

Best regards,

MrJackDuaner

**Comment by laruence** [comment]
hmm, seems I have to make a new release to trigger the PECL build cycle .... maybe I will do it next week (I am refactor Yaf for performance )

**Comment by MrJackDuaner** [comment]
Good afternoon,
Ok. Good news. Thank you.
Best regards,
MrJackDuaner

**Comment by laruence** [comment]
yaf-3.1.0 has been release at pecl: https://pecl.php.net/package/yaf   it shows 11% qps improvement while doing simple bench.
the windows dll for 7.4 should be show in hours, if not, I will make it happen, thanks

**Comment by MrJackDuaner** [comment]
Good morning,
Thank you very much for the sources / DLL.
I embedded php_yaf.dll in PHP 7.4 and I updated PHP handler on my local Web server (IIS) to use this new version.
My project uses Index controller by default.
Calling URL https://localhost/myproject/ should redirect to Auth controller for authentication from Index controller.
And when I call this URL in my browser, I can see the following errors in logs from log4php :

`2020-03-16T10:50:36+01:00 ERROR Failed opening controller script C:\wwwroot\myproject\application\controllers\Myproject.php: No such file or directory from /myproject/ ErrorController.ErrorAction(C:\wwwroot\myproject\application\controllers\Error.php:14)  
2020-03-16T10:50:36+01:00 ERROR Failed opening controller script C:\wwwroot\myproject\application\controllers\Myproject.php: No such file or directory from /myproject/Auth/ ErrorController.ErrorAction(C:\wwwroot\myproject\application\controllers\Error.php:14)`

As you can see the default behaviour has changed and it searches for Myproject controller.
We can see there is the redirection expected but errors are displayed and the page displayed in Web browser shows error about page not correctly redirected.
Do you have any idea on this new issue ?
Best regards,
MrJackDuaner

**Comment by laruence** [comment]
is /myproject a exisits folder ? please try to dump the request uri, and path_info

**Comment by laruence** [comment]
and which router do you use? yaf-route-static?

**Comment by laruence** [comment]
this should be fixed, I made a new release 3.1.2, please check with it while the windows build done(in hours). thanks

**Comment by MrJackDuaner** [comment]
Good morning,
You found and fixed the problem !
I've just integrated php_yaf.dll 3.1.2 with PHP 7.4.3 and it works like a charm. I have tested a complete navigation in the application and all is alright. There is no error in the output logs about YAF.
To answer your questions :
- Folder /myproject exists. Application was running fine with PHP 7.3.6 and YAF 3.0.8. I have just switched PHP handler to use PHP 7.4.3 and YAF 3.1.2 now.
- I use default router (_initRoute method empty in Bootstrap.php).
Thank you very much for your time on PHP and YAF.
Best regards,
MrJackDuaner
